pytorch_lightning/utilities/imports.py

```diff
@@ -107,4 +107,4 @@ def _fault_tolerant_enabled() -> bool:
     EXPERIMENTAL
     the `reset` function from `_MultiProcessingDataLoaderIter` was introduced in PyTorch 1.7 but we need to mock it.
     """
-    return _TORCH_GREATER_EQUAL_1_7 and bool(os.getenv("PL_FAULT_TOLERANT_TRAINING", False))
+    return _TORCH_GREATER_EQUAL_1_7 and int(os.getenv("PL_FAULT_TOLERANT_TRAINING", 0))
```

Comment:
took me a while to figure this one out. 
beware: `bool("0") is True`

Reply:
I added a test for this function, see below in `test_auto_restart.py`

Reply:
My bad! I broke it in a previous PR then